test/functional/feature_block.py

```diff
@@ -1338,13 +1340,17 @@ def sign_tx(self, tx, spend_tx):
             return
         sign_input_legacy(tx, 0, spend_tx.vout[0].scriptPubKey, self.coinbase_key)
 
-    def create_and_sign_transaction(self, spend_tx, value, script=CScript([OP_TRUE])):
-        tx = self.create_tx(spend_tx, 0, value, script)
+    def create_and_sign_transaction(self, spend_tx, value, output_script=None):
+        if output_script is None:
+            output_script = CScript([OP_TRUE])
```

Comment:
We are doing the same check for a `None` `output_script` in `create_tx`,
 should we instead delete this and rely on the check in `create_tx`?

```suggestion
```

After removing the check here and `next_block` the tests still passes

Reply:
Yes, but both are anyone-can-spend scripts. Is there a reason to prefer `CScript([OP_TRUE])` for `create_and_sign_transaction` and `next_block`? We can rely on the check in `create_tx` to avoid redundancy ?